plugins/cache_promote/lru_policy.cc

```diff
@@ -0,0 +1,186 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+#include <unistd.h>
+
+#include "lru_policy.h"
+
+#define MINIMUM_BUCKET_SIZE 10
+static LRUEntry NULL_LRU_ENTRY; // Used to create an "empty" new LRUEntry
+
+LRUPolicy::~LRUPolicy()
+{
+  TSDebug(PLUGIN_NAME, "LRUPolicy DTOR");
+  TSMutexLock(_lock);
```

Comment:
Why are you locking in the destructor?  Is the destructor modifying other objects? If some other thread is trying to touch this and just misses getting the lock, then instead it'll crash working with a dead object.